content_scripts/hints.js

```diff
@@ -98,9 +105,35 @@ var Hints = (function(mode) {
         lastMouseTarget = element;
     }
 
+
+    function cycle(dir) {
+        currentHintGroup += dir;
+        if(currentHintGroup < 0) {
+            currentHintGroup = hintGroupLen - 1;
+        } else if (currentHintGroup > hintGroupLen - 1) {
+            currentHintGroup = 0;
+        }
+        var hints = holder.find('>div');
+        hints.removeClass('inactive-group');
+        var selector = '[data-group="' + currentHintGroup + '"]';
+        hints.not(selector)
+            .addClass('inactive-group')
+            .each(function() {
+                var dist = $(this).data('group') - currentHintGroup;
+                $(this).text(dist);
+            });
+        hints.not('.inactive-group')
+            .each(function() {
+                $(this).text($(this).data('label'));
+            });
+    }
+
     function refresh() {
         var matches = [];
         var hints = holder.find('>div');
+        if (runtime.conf.hintGroups === true) {
```

Comment:
Since when `hintGroups` is on, hint labels will always be one letter, so this function `refresh` will be unnecessary. Therefore we don't need the change here.